### PR TITLE
Documented the new kernel.controller_arguments event

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | master (3.1+) |
| Fixed tickets | https://github.com/symfony/symfony/pull/18440 |

### CODE REVIEW

File: components/http_kernel/introduction.rst [hunk 1]
```diff
@@ -341,6 +343,13 @@ of arguments that should be passed when executing that callable.
        :class:`Symfony\\Component\\HttpFoundation\\Request` object, then the
        ``Request`` is passed in as the value.
 
+.. versionadded:: 3.1
+     The ``kernel.controller_arguments`` event was introduced in Symfony 3.1.
+
+Once controller arguments have been resolved by Symfony, the ``kernel.controller_arguments``
+event is triggered. This allows to modify or replace the arguments to fit your
```
Comment (stof) [inline components/http_kernel/introduction.rst]: it now also allows to replace the controller

File: reference/events.rst [hunk 2]
```diff
@@ -80,6 +80,33 @@ Listener Class Name
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 3.1
+     The ``kernel.controller_arguments`` event was introduced in Symfony 3.1.
+
+**Event Class**: :class:`Symfony\\Component\\HttpKernel\\Event\\FilterControllerArgumentsEvent`
+
+This event can be an entry point used to modify the arguments of the controller
+once they have been resolved by Symfony::
+
```
Comment (OskarStark) [inline reference/events.rst]: namespace is missing, otherwise its not very clear where to put this method IMO

Comment (stof) [inline reference/events.rst]: you can put it where you want (as long as you tell Symfony that this method is a listener for this event) .

And the current proposal is consistent with other snippets in the page

File: components/http_kernel/introduction.rst (hunk 1 above)
Comment (javiereguiluz) [inline components/http_kernel/introduction.rst]: I've reworded this. Thanks!

**Comment by javiereguiluz** [comment]
I've made the requested changes. Another PR ready to be merged. Thanks!

File: reference/events.rst [hunk 3]
```diff
@@ -80,6 +80,33 @@ Listener Class Name
...
 
+``kernel.controller_arguments``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 3.1
+     The ``kernel.controller_arguments`` event was introduced in Symfony 3.1.
+
+**Event Class**: :class:`Symfony\\Component\\HttpKernel\\Event\\FilterControllerArgumentsEvent`
+
+This event can be an entry point used to modify the arguments of the controller
```
Comment (xabbuh) [inline reference/events.rst]: I would remove "an entry point".

**Comment by xabbuh** [comment]
👍

File: reference/events.rst (hunk 3 above)
Comment (javiereguiluz) [inline reference/events.rst]: Fixed. Thanks.

**Comment by javiereguiluz** [comment]
After the last fixes, this is ready to be merged. Thanks!

**Comment by WouterJ** [comment]
This is probably the most confusing feature of Symfony 3.1 and that's already visible in this PR: Replacing arguments is *not* the main reason for this new event.

This new event is triggered after arguments are resolved, so some logic can be executed based on these arguments. For instance, the `@Security` annotation will resolve the expressions during this event (expressions can use the arguments).

If you want to replace arguments, writing a custom ArgumentValueResolver is the way to go (after that PR is merged, we should definitely reference it in this section).

**Comment by stof** [comment]
A use case to replace arguments is when you want to replace both the controller and the argument, after resolution of the original arguments (because your new controller is a closure wrapping the original controller). This was the use case brought by @fabpot (for a Blackfire use case IIRC) to allow replacing the controller in this event as well.

**Comment by WouterJ** [comment]
Status: Needs Work

**Comment by javiereguiluz** [comment]
@WouterJ I can't finish this PR. If you or any other volunteer can finish it, please take over this PR. Thanks.

**Comment by javiereguiluz** [comment]
Closing it because I can't finish it. I've opened an issue related to this: #6854.

**Pull request closed without merging**